embedded-cal-nrf54l15/src/lib.rs

```diff
@@ -0,0 +1,396 @@
+#![no_main]
+#![no_std]
+
+use defmt_rtt as _;
+use panic_probe as _;
+
+const BLOCK_SIZE: usize = 64;
+
+pub struct Nrf54l15Cal {
+    p: nrf54l15_app_pac::Peripherals,
+}
+
+impl Nrf54l15Cal {
+    pub fn new(p: nrf54l15_app_pac::Peripherals) -> Self {
+        // Enable cryptomaster
+        p.global_cracen_s.enable().write(|w| {
+            w.cryptomaster().set_bit();
+            w.rng().set_bit();
+            w.pkeikg().set_bit()
+        });
+
+        Self { p }
+    }
+}
+
+impl Drop for Nrf54l15Cal {
+    fn drop(&mut self) {
+        // Disable cryptomaster on drop
+        self.p.global_cracen_s.enable().write(|w| {
+            w.cryptomaster().clear_bit();
+            w.rng().clear_bit();
+            w.pkeikg().clear_bit()
+        });
+    }
+}
+
+#[repr(u8)]
+#[derive(PartialEq, Eq, Debug, Clone, Copy, defmt::Format)]
+pub enum HashAlgorithm {
+    Sha256 = 0x08,
+}
+
+impl embedded_cal::HashAlgorithm for HashAlgorithm {
+    fn len(&self) -> usize {
+        match self {
+            HashAlgorithm::Sha256 => 32,
+        }
+    }
+
+    // FIXME: See from_cose_number definition
+    fn from_cose_number(number: impl Into<i128>) -> Option<Self> {
+        match number.into() {
+            -10 => Some(HashAlgorithm::Sha256),
+            _ => None,
+        }
+    }
+
+    fn from_ni_id(number: u8) -> Option<Self> {
+        match number {
+            1 => Some(HashAlgorithm::Sha256),
+            _ => None,
+        }
+    }
+
+    fn from_ni_name(name: &str) -> Option<Self> {
+        match name {
+            "sha-256" => Some(HashAlgorithm::Sha256),
+            _ => None,
+        }
+    }
+}
+
+#[derive(defmt::Format)]
+pub struct HashState {
+    algorithm: HashAlgorithm,
+    state: Option<[u8; 32]>,
+    block: [u8; BLOCK_SIZE],
+    block_bytes_used: usize,
+    processed_bytes: usize,
+}
+
+pub enum HashResult {
+    Sha256([u8; 32]),
+}
+
+impl AsRef<[u8]> for HashResult {
+    fn as_ref(&self) -> &[u8] {
+        match self {
+            HashResult::Sha256(r) => &r[..],
+        }
+    }
+}
+
+#[repr(C)]
+#[derive(Debug, Clone, Copy, defmt::Format)]
+pub struct Descriptor {
+    pub addr: *mut u8,
+    pub next: *mut Descriptor,
+    pub sz: u32,
+    pub dmatag: u32,
```

Comment:
```suggestion
    addr: *mut u8,
    next: *mut Descriptor,
    sz: u32,
    dmatag: u32,
```

(Or: Why do these need to be pub?)

Reply:
Why do they even need to be pub(crate)? So far, all code that is accessing them is within the same module.